raster/r.watershed/testsuite/test_equal_ram_seg_output.py

```diff
@@ -0,0 +1,160 @@
+        """Ensures expected computational region and setup"""
+        # Always use the computational region of the raster elevation
+        cls.use_temp_region()
+        cls.runModule("g.region", raster=cls.elevation)
+
+        # random points raster
+        cls.runModule(
+            "r.random",
+            input=cls.elevation,
+            npoints=10000,
+            raster="random_fraction",
+            seed=1234,
+            overwrite=True,
+        )
+        cls.runModule(
+            "r.mapcalc",
+            expression="random_percent=random_fraction*100",
+            overwrite=True,
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """Remove the temporary region"""
+        cls.runModule("g.remove", flags="f", type="raster", name=cls.tmp_input_rasters)
+        cls.del_temp_region()
+
+    def tearDown(self):
+        """Remove the outputs created from the watershed module
+
+        This is executed after each test run.
+        """
+        self.runModule(
+            "g.remove",
+            flags="f",
+            type="raster",
+            pattern=",".join([o + "__*" for o in self.outputs]),
+        )
+
+    def same_ram_seg_output(self, outputs=None, **input_args):
+        """Check if the output of the ram and seg version is the same."""
+
+        outputs = outputs or self.outputs
+
+        flags = dict(ram="", seg="m")
+        kw = dict(
+            elevation=self.elevation,
+            threshold=1000,
+            overwrite=True,
+        )
+        kw.update(input_args)
+        # run module with/without -m
+        for n, f in flags.items():
+            # add outputs
+            kw.update({o: "%s__%s" % (o, n) for o in outputs})
+            kw["flags"] = input_args.get("flags", "") + f
+            self.assertModule("r.watershed", **kw)
+
+        # check difference of outputs
+        msg = "ram and seg version output %s is not the same"
+        msg += " with input " + str(input_args) if input_args else ""
+        passes = []
+        for o in outputs:
+            # subTest unfortunately doesnt work here
+            # with self.subTest("Testing difference in ram vs seg output", output=o):
+            prec = self.output_precision.get(o, 0)
+            try:
+                self.assertRastersNoDifference(
+                    "%s__ram" % o, "%s__seg" % o, prec, msg=msg % o
+                )
+                passes.append(True)
+            except AssertionError:
+                passes.append(False)
+        return passes
+
+    def test_same_ram_seg_output(self):
+
+        passes = []
+        for oi in self.inputs:
+            passes.append(self.same_ram_seg_output(**oi))
+
+        # create nice markdown table of matches
+        msg = (
+            "Output of ram and seg versions of r.watershed do not match:"
+            + "\n\n"
+            + self.md_table(passes)
+            + "\n"
+        )
+
+        self.assertTrue(all([all(p) for p in passes]), msg=msg)
```

Comment:
I think this test looks good. The only thing I would change is the number of reported failures. It's currently 1 from `AssertionError` if any tests fail. It would be useful to report the number of individual failed tests (red circles) somewhere.